testcases/kernel/syscalls/fallocate/fallocate03.c

```diff
@@ -1,277 +1,96 @@
+	{FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS + 2}
 };
 
-int TST_TOTAL = sizeof(test_data) / sizeof(test_data[0]);
-int block_size;
-int buf_size;
+static int block_size;
+static int buf_size;
 
-/******************************************************************************
- * Performs all one time clean up for this test on successful
- * completion,  premature exit or  failure. Closes all temporary
- * files, removes all temporary directories exits the test with
- * appropriate return code by calling tst_exit() function.
-******************************************************************************/
-void cleanup(void)
+static void cleanup(void)
 {
-	/* Close all open file descriptors. */
-	if (close(fd) == -1)
-		tst_resm(TWARN | TERRNO, "close(%s) failed", fname);
-
-	tst_rmdir();
-
+	if (fd)
+		SAFE_CLOSE(fd);
 }
 
-/*****************************************************************************
- * Performs all one time setup for this test. This function is
- * used to create temporary dirs and temporary files
- * that may be used in the course of this test
- ******************************************************************************/
-
-void setup(void)
+static void write_data_to_file(void)
 {
-	/* Create temporary directories */
-	TEST_PAUSE;
-
-	tst_tmpdir();
+	char buf[buf_size + 1];
 
-	sprintf(fname, "tfile_sparse_%d", getpid());
-	fd = SAFE_OPEN(cleanup, fname, O_RDWR | O_CREAT, 0700);
-	get_blocksize(fd);
-	populate_file();
-	file_seek(BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS);	/* create holes */
-	populate_file();
-	file_seek(0);		/* Rewind */
+	for (int blocks = 0; blocks < BLOCKS_WRITTEN; blocks++) {
+		for (int i = 0; i < buf_size; i++)
+			buf[i] = 'A' + (i % 26);
+		buf[buf_size] = '\0';
+		SAFE_WRITE(SAFE_WRITE_ALL, fd, buf, buf_size);
+	}
 }
 
-/*****************************************************************************
- * Gets the block size for the file system
- ******************************************************************************/
-void get_blocksize(int fd)
+static void setup(void)
 {
-	struct stat file_stat;
+	char fname[NAME_MAX];
 
-	if (fstat(fd, &file_stat) < 0)
-		tst_resm(TFAIL | TERRNO,
-			 "fstat failed while getting block_size");
+	sprintf(fname, "tfile_sparse_%d", getpid());
+	fd = SAFE_OPEN(fname, O_RDWR | O_CREAT, 0700);
+	struct stat file_stat;
 
+	// get fs block size
+	SAFE_FSTAT(fd, &file_stat);
 	block_size = (int)file_stat.st_blksize;
 	buf_size = block_size;
-}
 
-/*****************************************************************************
- * Create a Hole in the file
- ******************************************************************************/
-void file_seek(off_t offset)
-{
-	offset *= block_size;
-	lseek(fd, offset, SEEK_SET);
+	write_data_to_file();
+	// create holes
+	lseek(fd, block_size * (BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS), SEEK_SET);
+	write_data_to_file();
+	lseek(fd, 0, SEEK_SET); // Rewind
```

Comment:
Also please remove this comment and the one above (create holes), not much useful.